### PR TITLE
Semantics (Accessibility, Indexability)

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: sky/services/semantics/semantics.mojom [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+struct SemanticsNode {
+  uint32 id;
+  SemanticFlags? flags;
+  SemanticStrings? strings;
+  SemanticGeometry? geometry;
+  array<SemanticsNode> children;
+};
+
+struct SemanticFlags {
```
Comment (chinmaygarde) [inline sky/services/semantics/semantics.mojom]: The use of the plural form here indicates that the values must be used to form a mask.  Shouldn't this be something like:
````
struct SemanticFlags {
  canBeTapped =                1 << 0,
  canBeLongPressed =           1 << 1,
  canBeScrolledHorizontally =  1 << 2,
  canBeScrolledVertically =    1 << 3,
  hasCheckedState =            1 << 4,
  isChecked =                  1 << 5,
}
```
Unless my understanding of struct definitions in mojoms is completely off.

OTOH, if you don't want to create a mask, the `SemanticsNode` should probably have an array of these (`array<SemanticFlags>`). For example, something being both horizontally and vertically scrollable (array or mask, probably the latter given your comment on the packing)

Comment (chinmaygarde) [inline sky/services/semantics/semantics.mojom]: Ignore my previous comment on the struct definition. Read this as an enum :P

**Comment by chinmaygarde** [comment]
lgtm

**Comment by Hixie** [comment]
@abarth I'm landing this now so that by the time we next rev the engine I'll be ready to land the framework side.

**Pull request merged**